internal/cmd/kafka/command_clientconfig_create.go

```diff
@@ -186,38 +170,44 @@ func (c *createCommand) create(configId string, srApiAvailable bool) func(cmd *c
 	}
 }
 
-func (c *createCommand) setKafkaCluster(cmd *cobra.Command, configFile string) (string, error) {
+func (c *clientConfigCommand) setKafkaCluster(cmd *cobra.Command, configFile string) (string, error) {
 	// get kafka cluster from context or flags, including key pair
 	kafkaCluster, err := c.Config.Context().GetKafkaClusterForCommand()
 	if err != nil {
 		return "", err
 	}
 
+	if err := addApiKeyToCluster(cmd, kafkaCluster); err != nil {
+		return "", err
+	}
+
 	// Only validate that the key pair matches with the cluster if it's passed via the flag.
 	// This is because currently "api-key store" does not check if the secret is valid. Therefore, if users
 	// choose to use the key pair stored in the context, we should use it without doing a validation.
 	flagKey, flagSecret, err := c.Config.Context().KeyAndSecretFlags(cmd)
 	if err != nil {
 		return "", err
 	}
-	if flagKey != "" && flagSecret != "" {
+	if flagKey != "" || flagSecret != "" {
```

Comment:
`KeyAndSecretFlags` returns an error if the key is empty and the secret isn't, so I think we can just replace this with `if flagKey != "" {` since `false, true` seems unreachable.

Reply:
```suggestion
	if flagKey != "" {
```